### PR TITLE
Weaviate - throw error if storing vectors fails

### PR DESCRIPTION
**Pull request by the-powerpointer**
When storing vectors at Weaviate, different errors may occur. The current implementation already processes some of them, like connection reset or similar.
However, still when we get an API response, this does not mean that the insertion into the DB was successful.
Examples of such failures:

- store a document without having the schema created before
- store a document with some metadata of a type which is different than the current schema

For such errors we need to check the response from the API call.

The former implementation resulted in such cases in the weird status that the method `addVectors(..)` returned an array of ids, suggesting that the objects are available in Weaviate, but they are not. Furthermore no error is logged at all, so one don't even notice that and what went wrong.

This change collects all error messages from the API responses, and if there are any, lets the method `addVectors(..)` throw an Error containing those messages.

### CODE REVIEW

File: langchain/src/vectorstores/weaviate.ts [hunk 1]
```diff
@@ -164,12 +164,22 @@ export class WeaviateStore extends VectorStore {
     });
 
     try {
-      await this.client.batch
+      const responses = await this.client.batch
```
Comment (jacoblee93) [inline langchain/src/vectorstores/weaviate.ts:167]: So this was buggy before and `client.batch` doesn't throw at all?

File: langchain/src/vectorstores/weaviate.ts [hunk 2]
```diff
@@ -164,12 +164,22 @@ export class WeaviateStore extends VectorStore {
...
         .do();
+      // if storing vectors fails, we need to know why
+      const errorMessages: string[] = [];
+      responses.forEach((response) => {
+        if (response?.result?.errors?.error) {
+          errorMessages.push(...response.result.errors.error.map((err) => err.message ?? ""));
+        }
+      });
+      if (errorMessages.length > 0) {
+        throw new Error(errorMessages.join("\n"));
```
Comment (jacoblee93) [inline langchain/src/vectorstores/weaviate.ts:179]: Might be good to have some base case here if no messages can be parsed - e.g. `Unspecified error`

**Review by jacoblee93 — CHANGES_REQUESTED** [review]
Thanks for this! Can you also run `yarn format` and `yarn lint`?

File: langchain/src/vectorstores/weaviate.ts (hunk 1 above)
Comment (the-powerpointer) [reply langchain/src/vectorstores/weaviate.ts:167]: Ahm, no. As written in the PR description, certain errors were already catched correctly. Just try and address a server, which does not exist, e.g. `http://localhost:8080` -> that results in an error `FetchError: request to http://localhost:8080/v1/batch/objects failed, reason: connect ECONNREFUSED ::1:8080` also without the change of this PR.

However, if the server processes the request and sends a response, it still may contain an error. In my cases these were always 4xx errors, mostly 422 - Unprocessable Entity. However, in principle I could also imagine that Weaviate could respond with 5xx errors this way.
These kind of errors are catched with the changes of this PR.

File: langchain/src/vectorstores/weaviate.ts (hunk 2 above)
Comment (the-powerpointer) [reply langchain/src/vectorstores/weaviate.ts:181]: Yep, I didn't really think about a sensible fallback in line 175 and just took the empty string to fulfil typing restrictions.
Anyways, just preparing for this possibility I'd change the fallback in line 175 from empty string to `"!! Unfortunately no error message was presented in the API response !!"`. 
Commit follows.

**Comment by the-powerpointer** [comment]
Ideally I'd like to add some tests to prove that 4xx and 5xx errors are handled correctly. However, this requires some deep mocking of the Weaviate classes, and I'm unfortunately not an expert in JS/TS at all 😟, not to name deep mocking in Jest...

Anyway, what is the issue with deploying the preview? Is that anything I can influence? I saw that also other PRs fail with this error...

**Comment by jacoblee93** [comment]
Yeah we're having an issue with Vercel deploys. Don't worry about it, will merge tonight!

**Review by jacoblee93 — APPROVED** [review]

**Comment by jacoblee93** [comment]
Thanks!

**Pull request merged**